### PR TITLE
joyent/prr#23 editor session could be optional

### PR DESCRIPTION
**Pull request by jlevon**

### CODE REVIEW

File: lib/prr.js [hunk 1]
```diff
@@ -556,8 +527,7 @@ function editCommitMessage(arg, cb) {
  * message (usually the title of the PR) followed by the remaining lines.
  *
  * @param {String} args.commitMessagePath - The file path to read
- * @param {Function} cb - `function (err, first line of message,
- *                                   remainder of commit message)`
+ * @param {Function} cb - callback
```
Comment (timfoster) [inline lib/prr.js]: I'm not sure which style is more correct - have a method modify a context object, or make it explicit what values we're 'returning' via the callback.  It certainly makes for more concise code to just modify the context object.

File: lib/prr.js [hunk 2]
```diff
@@ -638,13 +611,13 @@ function yesNoPrompt(args, cb) {
...
     console.log('');
 
-    var user_question = 'Squash and merge this PR? (y/n, Ctrl-C to abort) ';
+    var user_question = 'Squash and merge this PR? ([y]es/[e]dit/[q]uit)';
     var prompt_schema = {
         properties: {
             answer: {
                 description: user_question,
-                pattern: /^[yn]$/,
-                message: 'y or n will do!',
```
Comment (timfoster) [inline lib/prr.js]: I'll be sad to see my Return Of The Jedi easter egg go. Will have to find another opportunity!

File: lib/prr.js [hunk 3]
```diff
@@ -656,15 +629,56 @@ function yesNoPrompt(args, cb) {
...
-                return;
+                args.abort = true;
+            } else if (result.answer[0] === 'y') {
+                args.commitMessageAccepted = true;
+            } else if (result.answer[0] === 'e') {
+                args.commitMessageAccepted = false;
+            } else if (result.answer[0] === 'q') {
+                args.abort = true;
+            } else {
+                console.log(JSON.stringify(result));
```
Comment (timfoster) [inline lib/prr.js]: Debugging code left in? Do we want to complain here?

File: lib/prr.js (hunk 1 above)
Comment (jlevon) [reply lib/prr.js]: Yeah, that's the main reason to have it like this IMO: it's much easier to see the pipeline. The only case you'd not do that would be callign out to something that's looser-coupled than these routines are IMO

File: lib/prr.js (hunk 2 above)
Comment (jlevon) [reply lib/prr.js]: Erk, sorry, I missed that :)

File: lib/prr.js (hunk 3 above)
Comment (jlevon) [reply lib/prr.js]: removed - this shouldn't happen now

**Pull request merged**